src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/TypeCast.cs

```diff
@@ -466,13 +465,10 @@ private static unsafe object CheckCastClassSpecial(MethodTable* pTargetType, obj
                     break;
             }
 
-            goto fail;
+            ThrowHelper.ThrowInvalidCastException();
 
         done:
             return obj;
-
-        fail:
-            return ThrowInvalidCastException(pTargetType);
```

Comment:
My old comment still stands - this is the existing structuring in the other copy of TypeCast.cs used with JIT.

They should be in sync, so either do the same change there (better to be 100% sure this will not deoptimize codegen) or  undo it here.